From 569251643feae73808ae6e491b1ba1ffb0375d6c Mon Sep 17 00:00:00 2001
From: Johannes Gallmann <gallmann@google.com>
Date: Mon, 22 May 2023 10:21:02 +0200
Subject: [PATCH] Fix PrivacyChip not visible issue

Bug: 281807669
Test: Manual, i.e. posting the following sequence of events (within few milliseconds) to the scheduler and observe the behaviour with and without the fix: Mic in use -> Mic not in use -> Mic in use
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:a45e1d045770eaabfdbf0e1212c9eb84caf1d565)
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:20ea049a4a52dbc8d4e5ed957a2b6b9aa02a2f34)
Merged-In: I9851e6ed4cb956d0459ef56251eb0ef3210764b8
Change-Id: I9851e6ed4cb956d0459ef56251eb0ef3210764b8
---
 .../src/com/android/systemui/statusbar/events/StatusEvent.kt  | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/events/StatusEvent.kt b/packages/SystemUI/src/com/android/systemui/statusbar/events/StatusEvent.kt
index d4d84c138b20..f610101631dd 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/events/StatusEvent.kt
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/events/StatusEvent.kt
@@ -86,9 +86,7 @@ class PrivacyEvent(override val showAnimation: Boolean = true) : StatusEvent {
     }
 
     override fun shouldUpdateFromEvent(other: StatusEvent?): Boolean {
-        return other is PrivacyEvent &&
-                (other.privacyItems != privacyItems ||
-                other.contentDescription != contentDescription)
+        return other is PrivacyEvent
     }
 
     override fun updateFromEvent(other: StatusEvent?) {
-- 
2.41.0.255.g8b1d071c50-goog

